src/core/crypto_aos.c

```diff
@@ -1249,6 +1249,13 @@ int32_t Crypto_AOS_ProcessSecurity(uint8_t* p_ingest, uint16_t len_ingest, uint8
 
     }
 
+    // check sa state before decrypting
+    if (sa_ptr->sa_state != SA_OPERATIONAL)
+    {
+        printf(KRED "Error: SA Not Operational \n" RESET);
```

Comment:
Can you wrap these in #ifdef DEBUG cases?  Any print we supply should only be used in Debug cases.